### PR TITLE
Updating endpoints to return certified collections

### PR DESCRIPTION
**Pull request by romanblanco**
Fixes: https://github.com/ansible/galaxy-dev/issues/132

cc/ @newswangerd @cutwater

### CODE REVIEW

File: galaxy_api/api/v3/viewsets.py [hunk 1]
```diff
@@ -94,6 +99,37 @@ def retrieve(self, request, *args, **kwargs):
...
+    @drf_action(
+        methods=["PUT"],
+        detail=True,
+        url_path="certified",
+        permission_classes=api_settings.DEFAULT_PERMISSION_CLASSES + [
+            permissions.IsPartnerEngineer
+        ],
+    )
+    def set_certified(self, request, *args, **kwargs):
+        namespace = self.kwargs['namespace']
```
Comment (newswangerd) [inline galaxy_api/api/v3/viewsets.py]: Can you update the request to only require `certification` and not `namespace`, `name` and `version`?

Those values are already provided in the URL and adding them to the request body is redundant.

Comment (cutwater) [reply galaxy_api/api/v3/viewsets.py]: I don't think these values are provided via request body.
`self.kwargs` consists of dictionary of URL path parameters.

Comment (cutwater) [reply galaxy_api/api/v3/viewsets.py]: +1

File: galaxy_api/api/ui/viewsets/collection.py [hunk 2]
```diff
@@ -145,36 +146,6 @@ def retrieve(self, request, *args, **kwargs):
         data = serializers.CollectionVersionDetailSerializer(response.results[0]).data
         return Response(data)
 
-    @drf_action(
```
Comment (cutwater) [inline galaxy_api/api/ui/viewsets/collection.py]: Certification API will likely change in the future. Let's keep it under `_ui`

File: tests/api/test_api_ui_collection_viewsets.py [hunk 3]
```diff
@@ -14,29 +14,3 @@ def setUp(self):
         patcher = mock.patch("galaxy_pulp.GalaxyCollectionVersionsApi", spec=True)
         self.versions_api = patcher.start().return_value
         self.addCleanup(patcher.stop)
-
```
Comment (cutwater) [inline tests/api/test_api_ui_collection_viewsets.py]: Please write unit tests for your changes.

**Comment by newswangerd** [comment]
When I log in with a non-partner engineer user I can still update the certification status on all the collections. I just have to submit the correct PUT request and the API response with the correct `Forbidden` response, but then continues to update the certification anyway.

File: galaxy_api/api/v3/viewsets.py (hunk 1 above)
Comment (romanblanco) [reply galaxy_api/api/v3/viewsets.py]: :+1:

**Comment by romanblanco** [comment]
> When I log in with a non-partner engineer user I can still update the certification status on all the collections. I just have to submit the correct PUT request and the API response with the correct `Forbidden` response, but then continues to update the certification anyway.

after conversation with @cutwater, seems that these changes need to be tested with user in `system:partner-engineers` group

File: galaxy_api/api/ui/viewsets/collection.py [hunk 4]
```diff
@@ -143,32 +148,35 @@ def retrieve(self, request, *args, **kwargs):
...
     def set_certified(self, request, *args, **kwargs):
-        namespace, name = self.kwargs['collection'].split('/')
-        version = self.kwargs['version']
-
+        namespace = self.kwargs['namespace']
         namespace_obj = get_object_or_404(models.Namespace, name=namespace)
         self.check_object_permissions(request, namespace_obj)
 
         api = galaxy_pulp.GalaxyCollectionVersionsApi(pulp.get_client())
+        serializer = CollectionVersionSerializer(data=request.data, context={'request': request})
```
Comment (alikins) [inline galaxy_api/api/ui/viewsets/collection.py]: CollectionVersionSerializer doesn't seem to be imported anywhere.

Comment (alikins) [reply galaxy_api/api/ui/viewsets/collection.py]:
``` diff
-        serializer = CollectionVersionSerializer(data=request.data, context={'request': request})
+        serializer = serializers.CollectionVersionSerializer(data=request.data,
+                                                             context={'request': request})
```

**Comment by alikins** [comment]
> after conversation with @cutwater, seems that these changes need to be tested with user in system:partner-engineers group


@romanblanco https://github.com/ansible/galaxy-api/pull/112/files#diff-a9698a4900b4d0380ae707a147e225f2  has some examples of unit tests that poke at /_ui/ .
And similar non-ui tests at https://github.com/ansible/galaxy-api/pull/107/files#diff-68e2a5273407c975295ba2c2db9940f8

File: galaxy_api/api/ui/urls.py [hunk 5]
```diff
@@ -15,6 +16,17 @@
...
+    'collections/(?P<collection>[0-9a-z_]+/[0-9a-z_]+)/versions/[0-9A-Za-z.+-]+/certified',
+    # 'collections/(?P<collection>{})/versions/(?P<version>{})/certified'.format(
+    #     viewsets.CollectionViewSet.lookup_value_regex,
+    #     viewsets.CollectionVersionViewSet.lookup_value_regex
+    # ),
+    # https://github.com/encode/django-rest-framework/issues/1337
+    # urlpatterns = format_suffix_patterns(urlpatterns)
+    viewsets.CollectionVersionViewSet,
+    basename='collection-versions',
+)
```
Comment (chouseknecht) [inline galaxy_api/api/ui/urls.py]: This should look something more like the following:

```
router.register(
    'collections/(?P<collection>{})/versions/(?P<version>{})/certified'.format(
        viewsets.CollectionViewSet.lookup_value_regex,
        utils.VERSION_REGEXP
    ),
    viewsets.CollectionVersionViewSet,
    basename='collection-versions',
)
```

Comment (cutwater) [reply galaxy_api/api/ui/urls.py]: `VERSION_REGEXP` is a standalone object and is not designed to be injected into routing expression.

Comment (cutwater) [reply galaxy_api/api/ui/urls.py]: Reusing `CollectionViewSet.lookup_value_regexp` is probably better though.

Comment (chouseknecht) [reply galaxy_api/api/ui/urls.py]: Per conversation in Slack with @cutwater, we don't need to register this at all. It's handled automatically by the `drf_action` decorator.

**Comment by romanblanco** [comment]
Pull Request updated, works. 

cc/ @cutwater 

> When I log in with a non-partner engineer user I can still update the certification status on all the collections. I just have to submit the correct PUT request and the API response with the correct `Forbidden` response, but then continues to update the certification anyway.

@newswangerd I didn't test this yet. Keeping the `[wip]` commit with `FIXME` note for now.

File: tests/api/test_api_ui_collection_viewsets.py (hunk 3 above)
Comment (romanblanco) [reply tests/api/test_api_ui_collection_viewsets.py]: @cutwater Unit tests will be added in https://github.com/ansible/galaxy-api/pull/121 so this PR is not blocked.

I'll rebase #121 after this PR is merged.

**Pull request merged**